Google.Cloud.AspNetCore/Google.Cloud.AspNetCore.Firestore.DistributedCache/FirestoreCacheGarbageCollector.cs

```diff
@@ -0,0 +1,79 @@
+﻿// Copyright (c) 2019 Google LLC.
+// 
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not
+// use this file except in compliance with the License. You may obtain a copy of
+// the License at
+// 
+// http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+// License for the specific language governing permissions and limitations under
+// the License.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Google.Cloud.Firestore;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+
+namespace Google.Cloud.AspNetCore.Firestore.DistributedCache
+{
+    public class FirestoreCacheGarbageCollector : BackgroundService
+    {
+        private readonly FirestoreCache _cache;
+        private readonly TimeSpan _frequency;
+        private readonly Random _random = new Random();
+        private readonly ILogger<FirestoreCacheGarbageCollector> _logger;
+
+
+        public FirestoreCacheGarbageCollector(FirestoreCache cache,

```

Comment:
Possibly accept an IScheduler? That allows a testable version of `Task.Delay`.

Reply:
Done.